src/Services/Identity/Identity.BusinessLogic/Services/Implementations/UserService.cs

```diff
@@ -0,0 +1,195 @@
+﻿using AutoMapper;
+using Identity.BusinessLogic.DTOs.RequestDTOs.User;
+using Identity.BusinessLogic.DTOs.ResponseDTOs;
+using Identity.BusinessLogic.Errors;
+using Identity.BusinessLogic.Services.Interfaces;
+using Identity.DataAccess.Entities;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.EntityFrameworkCore;
+using Shared;
+using Shared.Constants;
+using Shared.Wrappers;
+
+namespace Identity.BusinessLogic.Services.Implementations;
+
+public class UserService(
+    UserManager<User> _userManager,
+    RoleManager<Role> _roleManager,
+    IMapper _mapper) : IUserService
+{
+    public async Task<Response<UserDto>> CreateUserAsync(RegisterUserDto dto, CancellationToken cancellationToken)
+    {
+        var existingUser = await _userManager.FindByNameAsync(dto.UserName);
+        
+        if (existingUser is not null)
+        {
+            return Response.Failure<UserDto>(DomainErrors.User.UsernameConflict);
+        }
+
+        var user = _mapper.Map<User>(dto);
+        user.SecurityStamp = Guid.NewGuid().ToString();
+
+        var result = await _userManager.CreateAsync(user, dto.Password);
+        
+        if (!result.Succeeded)
+        {
+            return Response.Failure<UserDto>(new Error(
+                result.Errors.First().Code,
+                result.Errors.First().Description));
+        }
+
+        await _userManager.AddToRoleAsync(user, Roles.Client);
+
+        return _mapper.Map<UserDto>(user);
+    }
+
+    public async Task<Response<UserDto>> UpdateUserAsync(UpdateUserDto dto, CancellationToken cancellationToken)
+    {
+        var user = await _userManager.FindByIdAsync(dto.Id.ToString());
+        
+        if (user is not { DateDeleted: null })
+        {
+            return Response.Failure<UserDto>(DomainErrors.User.UserNotFoundById);
+        }
+        
+        user.UserName = dto.UserName;
+        user.Email = dto.Email;
+
+        var result = await _userManager.UpdateAsync(user);
+
+        return result.Succeeded
+            ? _mapper.Map<UserDto>(user)
+            : Response.Failure<UserDto>(new Error(
+                result.Errors.First().Code,
+                result.Errors.First().Description));
+    }
+
+    public async Task<Response> DeleteUserByIdAsync(int id)
+    {
+        var user = await _userManager.FindByIdAsync(id.ToString());
+        
+        if (user is not { DateDeleted: null })
+        {
+            return Response.Failure(DomainErrors.User.UserNotFoundById);
+        }
+
+        user.DateDeleted = DateTimeOffset.UtcNow;
+        var result = await _userManager.UpdateAsync(user);
+
+        return result.Succeeded
+            ? Response.Success()
+            : Response.Failure(new Error(
+                result.Errors.First().Code,
+                result.Errors.First().Description));
+    }
+
+    public async Task<Response> AddUserToRoleAsync(int userId, int roleId)
+    {
```

Comment:
A dto object must be passed to the service methods as an input parameter. Change everywhere